### PR TITLE
Add visual indicators of problems and details of import file

### PR DESCRIPTION
**Pull request by cobbspur**
Issue #5422
- change db.api response to post
- show details of import file
- show problems with import file
- remove notifications

### CODE REVIEW

File: core/client/app/controllers/settings/labs.js [hunk 1]
```diff
@@ -30,12 +32,13 @@ export default Ember.Controller.extend(Ember.Evented, {
...
-                notifications = this.get('notifications');
-
+                formData = new FormData();
             this.set('uploadButtonText', 'Importing');
             this.set('importErrors', '');
-            notifications.closePassive();
+            this.set('importDetails', '');
+            this.set('importStatus', '');
+            this.set('importProblemsPosts', '');
+            this.set('importProblemsTags', '');
```
Comment (novaugust) [inline core/client/app/controllers/settings/labs.js]: For setting lots of properties like this, you might use `this.setProperties`

```js
this.setProperties({
  importDetails: '',
  importStatus: '',
  importProblemsPosts: '',
  importProblemsTags: ''
});
```

**Comment by cobbspur** [comment]
Yeah I was looking to clean that whole thing up, I wanted to create an object and loop through those settings but I couldn't figure it for some reason, tried the usual ways but that is a little cleaner I suppose unless you have thoughts on that.

**Comment by jaswilli** [comment]
I think Import would make a really great component.  It seems like everything about it could be moved off the controller and encapsulated in a single component instead of being spread out between a controller, the controller's template and a partial.

**Comment by cobbspur** [comment]
That's interesting I have another piece of work I have been playing with where the importer is in a modal which more for the purpose of isolating some of the code and making it easier to pick through. It generally worked although there were occasionally some oddities I could not figure out. 

So far as the component idea goes, I am working on a component related issue at the moment, renaming some components and removing the popover and popover-button components that are not used currently so I could add it to that PR. There is already a gh-file-upload component I am renaming gh-import or gh-file-import so I guess it could live there.

**Comment by cobbspur** [comment]
I've updated the PR, I am going to use unloadAll to clear the store for now following the conversation in slack. Using the window.location is very un-Ghost-like. I will keep moving Import into a component of its own as a separate piece of work, possibly along with the component renaming

**Comment by jaswilli** [comment]
> Using the window.location is very un-Ghost-like.

No it's not.  It's the best way to reset an Ember application.

> I am going to use unloadAll

After this occurs, are you able to create a new post?  How about delete a user or change a user's role?

**Comment by cobbspur** [comment]
Damned linting
It is un- Ghost-like in the way the page is reloaded rather than a transition

So far as the other items there, the user role cannot be changed as it couldn't before, I have raised a separate issue on that one but I will test creating a new post I hadn't tried that

**Comment by novaugust** [comment]
We're doing the window.location somewhere else.. something to do with
login?
Anyways, reloading the page seems appropriate when you've just reset the
entire database, and like the safest thing to do, unless you follow the
unloadAll with a signOut and transitionTo signin?

**Comment by cobbspur** [comment]
Doing it with respect to the login is one thing as you are outside of the admin panel anyway. Reloading the page after an import looks and feels wrong, I would urge you both to try it out for yourselves, it is just one line of code after all, then decide on whether the feel of Ghost is natural and consistent with everything else in the admin-UI. This may still be something that lives in a modal as we talked about on slack with the confirm button clearly indicating to the user that there be a reload of some sort. But while labs is on the one page I am not in favour of reloading. There also the question here for me as to whether or not we are simply overlooking another bug by switching approach. If the intended function of unloadAll() or unloadAll(model) is not working then should we not be asking why?

**Comment by jaswilli** [comment]
> Reloading the page after an import looks and feels wrong

LOL.  You don't think I've tried it?  It "looks and feels" completely appropriate after just giving the application a new universe to exist in.

Just unloading the store and leaving the application in an indeterminate state is completely lazy.  If making the import behavior not break the app needs some design work, then let's get it done.

> unless you follow the
> unloadAll with a signOut and transitionTo signin?

Which is a hard refresh... :smile:

**Comment by cobbspur** [comment]
Yeah that's fine (the UI-redesign, there is an epic Issue #5422 where these sorts of suggestions would be really welcome but for the function of this PR, I am trying to add the first piece of gloss to a Labs item. The unloadAll is not new and there will not be any new bug. Implementing any other changes like going for a modal is kinda blocked for me by this one.

One other thing I had been wondering about as an alternative was whether there was a way to set a flag or property in the controller when there has been a successful import and then force the next transition to be a full refresh - that may just not be a thing but perhaps you know of something like that

**Comment by ErisDS** [comment]
With #5514 merged, the issue was blocking this from moving forward has a resolution. If you get this rebased it should be good to merge?

**Comment by cobbspur** [comment]
closing this for now. Will reopen when ready for merging

**Pull request closed without merging**